cmake/extensions.cmake

```diff
@@ -2261,3 +2263,368 @@ function(target_byproducts)
+
+  set(${var} ${${var}} PARENT_SCOPE)
+endfunction()
+
+# Usage:
+#   dt_node_exists(<var> NODE <node>)
+#
+# Does a node path refer to a node?
+#
+# Tests whether a node path refers to a node which exists, i.e. is defined
+# in the devicetree.
+#
+# the result of the check will be returned in the <var> parameter.
+#
+# <var>       : Return variable where the check result will be returned
+# NODE <node> : Node path
+function(dt_node_exists var)
+  set(req_single_args "NODE")
+  cmake_parse_arguments(DT_NODE "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_node_existsl(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_NODE_${arg})
+      message(FATAL_ERROR "dt_node_exists(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_NODE|${DT_NODE_NODE}")
+
+  if(${var})
+    set(${var} ${${var}} PARENT_SCOPE)
+  else()
+    set(${var} FALSE PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Usage:
+#   dt_node_has_status(<var> NODE <node> STATUS <status>)
+#
+# Tests whether a node path refers to a node which:
+# - exists in the devicetree, and
+# - has a status property matching the second argument
+#   (except that either a missing status or an “ok” status in the devicetree
+#    is treated as if it were “okay” instead)
+#
+# the result of the check will be returned in the <var> parameter.
+#
+# <var>           : Return variable where the check result will be returned
+# NODE <node>     : Node path
+# STATUS <status> : Status to check
+function(dt_node_has_status var)
+  set(req_single_args "NODE;STATUS")
+  cmake_parse_arguments(DT_NODE "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_node_has_status(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_NODE_${arg})
+      message(FATAL_ERROR "dt_node_has_status(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  dt_node_exists(${var} NODE ${DT_NODE_NODE})
+  if(NOT ${${var}})
+    set(${var} FALSE PARENT_SCOPE)
+  endif()
+
+  dt_prop(${var} NODE ${DT_NODE_NODE} PROPERTY status)
+
+  if(NOT DEFINED ${var} OR "${${var}}" STREQUAL ok)
+    set(${var} okay)
+  endif()
+
+  if(${var} STREQUAL ${DT_NODE_STATUS})
+    set(${var} TRUE PARENT_SCOPE)
+  else()
+    set(${var} FALSE PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Usage:
+#   dt_prop(<var> NODE <node> PROPERTY <prop>)
+# Get a devicetree property value.
+#
+#   dt_prop(<var> NODE <node> INDEX <idx> PROPERTY <prop>)
+# Get the value at index <idx> in an array type property.
+# If index is omitted and the property is of array type, then the value at
+# index 0 will be returned.
+#
+# the value will be returned in the <value> parameter.
```

Comment:
what is `<prop>` doesnt exist, what is returned?

Reply:
Then the `<var>` will be undefined.

If it exists, then `<var>` is defined with the value of the property.

Reply:
> Then the `<var>` will be undefined.

Docs?

Reply:
Function description improved